src/mesh/gmsh_reader.cpp

```diff
@@ -438,10 +501,117 @@ unsigned int ij_to_num(const unsigned int i,
     return i + j*n_per_line;
 }
 
+/**
+ * Solely used in the 3D case. Helps finding the face_nodes during recursive call.
+ * @param degree
+ * @return
+ */
+std::vector<unsigned int>
+face_node_finder(const unsigned int degree, const bool mesh_reader_verbose_output)
+{
+
+    /**
+    * For Debug output
+    */
+    const int mpi_rank = dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+    dealii::ConditionalOStream pcout(std::cout, mpi_rank==0);
+
+    // number of support points in each direction
+    const unsigned int n = degree + 1;
+
+    const unsigned int dofs_per_cell = dealii::Utilities::fixed_power<2>(n);
+
+    std::vector<unsigned int> h2l_2D(dofs_per_cell);
+    if (degree == 0) {
```

Comment:
That is a good point, the degree = 0 is a bit ambiguous. But, it will get executed when we have a second order quad/face with 9 points, then, once we call "face_node_finder", there will be only one point left on the inside of that face, which will return a position of 0 for it.

Reply:
Will be resolved by issue #232